crates/bevy_ui/src/ui_node.rs

```diff
@@ -238,15 +238,19 @@ pub enum ValArithmeticError {
 
 impl Val {
     /// A convenience function for simple evaluation of [`Val::Percent`] variant into a concrete [`Val::Px`] value.
-    /// Returns a [`ValArithmeticError::NonEvaluateable`] if the [`Val`] is impossible to evaluate into [`Val::Px`].
-    /// Otherwise it returns an [`f32`] containing the evaluated value in pixels.
+    /// Returns a [`ValArithmeticError::NonEvaluateable`] if the [`Val`] is impossible to resolve into [`Val::Px`].
+    /// Otherwise it returns an [`f32`] containing the resolved value in pixels.
     ///
-    /// **Note:** If a [`Val::Px`] is evaluated, it's inner value returned unchanged.
-    pub fn evaluate(&self, size: f32) -> Result<f32, ValArithmeticError> {
+    /// **Note:** If a [`Val::Px`] is resolved, it's inner value is returned unchanged.
+    pub fn resolve(self, size: f32, viewport_size: Vec2) -> Result<f32, ValArithmeticError> {
```

Comment:
Perhaps the `size` parameter should be renamed to `percentage_size` or `parent_size`? As `viewport_size` is also a `size`...

Reply:
Could even call this `resolve_to_px` if we wanted to be really explicit (although I think `resolve` is fine).

Reply:
Completely forgot about updating the docs for this one.

Reply:
`parent_size` is ok, but it doesn't feel super satisfying. Maybe `parent_length` might be better? We've mostly only used size to refer to 2d properties in the past.

Reply:
`parent_size` is good enough for now, can always be changed later.